pages/upgrade-guide/1.5.md

```diff
@@ -0,0 +1,306 @@
+
+We did a complete rethink of how auto enhancment with jQuery should work and the new module is FAST depending on the number of elements needing methods called its actually faster then directly calling each
+as the selection from the DOM is done only once per call. To make this possible and to expand past just widgets we have decided to make to major changes.
+
+First we have deprecated the ability to set a custom name space for data-attributes. jQuery mobile will now allways use the `ui` name space on all data-attributes.
+
+Old API:
+```html
+<div data-role="page">
+```
+
+New API:
+```html
+<div data-ui-role="page">
+```
+
+The second change is all widgets roles will now follow the patter `data-ui-role="method"` where method can be any valid jQuery method. so things like all inputs with a type of `text` being automaticlly enhanced will no longer happen you will need to add `data-ui-role="textinput"` to the input
+
+Old API:
+```html
+<input type="text" id="awesome-input" name="awesome-input">
+<input type="range" id="awesome-range" name="awesome-range">
+```
+
+New API:
+```html
+<input type="text" data-ui-role="textinput" id="awesome-input" name="awesome-input">
+<input type="number" data-ui-role="slider" id="awesome-range" name="awesome-range">
+```
+
+## Degrade Inputs
+
+In 1.4 and prior versions to promote progressive enhancment we had certain input types "auto degrade" a search to a text and a range to a number for example. The reason for this was to avoid the native html 5  control rendering so we can use the custom widget. There was one major problem with this which was that you cant just change the type attribute on an input in Internet Explorer. This means we needed to actually create a new element and replace it. This is a very destructive process because any data which was previously set or event listeners will be lost as the element has been replace.
+
+Starting in jQuery Mobile 1.5 this behivor is deprecated. You will now need to use the proper input type for the widget you are initalizeing. The only widgets which are effected by this change are textinput with a type of search and slider.
+
+Old API:
+```html
+<input type="search" id="awesome-input" name="awesome-input">
+<input type="range" id="awesome-range" name="awesome-range">
+```
+
+New API:
+```html
+<input type="text" data-ui-role="textinput" data-ui-type="search" id="awesome-input" name="awesome-input">
+<input type="number" data-ui-role="slider" id="awesome-range" name="awesome-range">
+```
+
+## Keep Native
+
+In 1.4 `$.mobile.keepNative` allowed you to set a selector which you could add to avoid auto enhancment. Now that all auto initialization is explicit this option is no longer needed and will be deprecated. There is no replacment for this option other then to just not add a `data-ui-role` to elements you dont want enhanced.
```

Comment:
In 1.4 `$.mobile.keepNative` allowed you to set a selector which you could add to avoid [~~auto enhancment~~ autoenhancement or auto-enhancement or automatic enhancement]. Now that all auto initialization is explicit this option is no longer needed and will be deprecated. There is no replac[e]ment for this option other th[~~e~~a]n to just not add a `data-ui-role` to elements you don[']t want enhanced.